cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -70,5 +90,95 @@ public void run() {
     long completionTime = System.currentTimeMillis();
     LOG.info("File cleanup completed, Successful file deletes - {}. Failed file deletes - {}. Log Cleanup took {} ms",
              deleteCount, failureCount, (completionTime - startTime));
+
+    return folderCleanupCount < folderCleanupBatchSize ? DEFAULT_DELAY_IN_MILLIS : FOLDER_CLEANUP_DELAY_IN_MILLIS;
+  }
+
+  private void cleanupLogFolders() {
+    LOG.info("Starting log folder cleanup");
+    try {
+      long startTime = System.currentTimeMillis();
+      LOG.debug("Starting log folder cleanup");
+      folderCleanupCount = 0;
+
+      // TODO: (CDAP-19437): Exclude recent log folders during cleanups in LogCleaner service.
+      // For now exclude folders for current date and current date - 1.
+      long currentTime = System.currentTimeMillis();
+      String currentDate = new SimpleDateFormat(LogFileManager.FOLDER_DATE_FORMAT).format(new Date(currentTime));
+      long prevTime = System.currentTimeMillis() - MILLISECONDS_IN_DAY;
```

Comment:
Why it is hardcoded to a day?

Reply:
Generalized this for N number of days. Defined const for the number of days to exclude during clean up.